src/HeatCondEquationSystem.C

```diff
@@ -230,6 +232,18 @@ HeatCondEquationSystem::register_nodal_fields(
                                stk::topology::NODE_RANK);
 
     copyStateAlg_.push_back(theCopyAlgA);
+
+    if ( numVolStates <= 2 ) return;
+
+    ScalarFieldType &dualNdVolN = dualNodalVolume_->field_of_state(stk::mesh::StateN);
+    ScalarFieldType &dualNdVolNp1 = dualNodalVolume_->field_of_state(stk::mesh::StateNP1);
+
+    CopyFieldAlgorithm *theCopyAlgDlNdVol
+      = new CopyFieldAlgorithm(realm_, part,
+                               &dualNdVolNp1, &dualNdVolN,
+                               0, 1,
+                               stk::topology::NODE_RANK);
+    copyStateAlg_.push_back(theCopyAlgDlNdVol);
```

Comment:
For BDF2, the number of states will be 3. Should also probably copy to `StateNM1`.

Reply:
@sayerhs Throughout the code we copy fields only from NP1 to N and never N to NM1 for a restart. I was under the impression that was because the restart files save states only at NP1. Is that not correct?